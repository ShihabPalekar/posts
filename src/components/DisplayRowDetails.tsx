import { Button } from "@mui/material";

type Props = {
  selectedRow: any;
  navigate: any;
};

const DisplayRowDetails: React.FC<Props> = ({ ...props }) => {
  const goBack = () => {
    props.navigate(-1);
  };
  return (
    <div style={{ margin: "20px" }}>
      <div>
        <pre role="json-data">{JSON.stringify(props.selectedRow, null, 2)}</pre>
      </div>
      <div>
        <Button variant="text" role="back-button" onClick={goBack}>
          GO BACK
        </Button>
      </div>
    </div>
  );
};

export default DisplayRowDetails;
