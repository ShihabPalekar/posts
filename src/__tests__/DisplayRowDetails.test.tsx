import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import DisplayRowDetails from "../components/DisplayRowDetails";
import { fireEvent, render, screen } from "@testing-library/react";

test("Row details page renders correctly", () => {
  const component = renderer.create(
    <DisplayRowDetails navigate={undefined} selectedRow={undefined} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Input Form component", () => {
  test("renders row-details component", () => {
    render(<DisplayRowDetails navigate={() => {}} selectedRow={undefined} />);
  });

  test("back button exists", () => {
    render(<DisplayRowDetails navigate={() => {}} selectedRow={undefined} />);
    expect(screen.getByRole("back-button")).toBeInTheDocument();
  });

  test("data is displayed", () => {
    render(<DisplayRowDetails navigate={() => {}} selectedRow={undefined} />);
    expect(screen.getByRole("json-data")).toBeInTheDocument();
  });

});
