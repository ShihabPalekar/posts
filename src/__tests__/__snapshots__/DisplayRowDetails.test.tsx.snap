// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row details page renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "20px",
    }
  }
>
  <div>
    <pre
      role="json-data"
    />
  </div>
  <div>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="back-button"
      tabIndex={0}
      type="button"
    >
      GO BACK
    </button>
  </div>
</div>
`;
